src/main/java/org/myrobotlab/service/Webcam.java

```diff
@@ -1,49 +1,225 @@
+  
+  protected int requestedFps = 30;
+
+  protected int frameIndex = 0;
+
+  protected String type = "jpg";
+
+  protected Integer height = 480;
+
+  long startSampleTs = 0;
+  
+  int lastFrameIndex = 0;
+
+  protected Integer port = 8080;
+
+  protected double quality = 0.9;
+
+  transient private WebcamStreamer streamer;
+
+  protected boolean useWebcamStreamer = false;
+
+  transient com.github.sarxos.webcam.Webcam webcam;
+
+  protected List<String> webcams = null;
+
+  protected Integer width = 640;
+
+  transient Thread worker = null;
 
   public Webcam(String n, String id) {
     super(n, id);
+    Platform platform = Platform.getLocalInstance();
+    if (platform.isArm() && platform.isLinux()) {
+      com.github.sarxos.webcam.Webcam.setDriver(new V4l4jDriver());
+    }
+    getWebcams();
   }
 
-  public void start() {
-    start(null);
+  public void capture() {
+    capture(null, null, null, null, null, null);
   }
 
-  public void start(Integer port) {
+  public synchronized void capture(int index, String type, Integer fps, Integer width, Integer height, Double quality) {
+    if (webcam != null) {
+      stopCapture();
+    }
+
+    if (fps != null) {
+      this.requestedFps = fps;
+    }
+
+    if (width != null) {
+      this.width = width;
+    }
+    
+    if (type != null) {
+      this.type = type;
+    }
+
+    if (height != null) {
+      this.height = height;
+    }
+
+    if (quality != null) {
+      this.quality = quality;
+    }
+
+    List<com.github.sarxos.webcam.Webcam> cams = com.github.sarxos.webcam.Webcam.getWebcams();
+    if (index >= cams.size()) {
+      error("camera index requested %d but only %d cameras", index, cams.size());
+      return;
+    }
+
+    // webcam = com.github.sarxos.webcam.Webcam.getWebcamByName(name);
+    webcam = cams.get(index);
+    selectedCamera = webcam.getName();
+
     if (webcam == null) {
       webcam = com.github.sarxos.webcam.Webcam.getDefault();
+      selectedCamera = webcam.getName();
+      info("starting webcam on port %d", port);
     }
-    if (port == null) {
-      port = 8080;
+
+    webcam.setViewSize(new Dimension(this.width, this.height));
+
+    if (useWebcamStreamer && streamer == null) {
+      info("starting webstreamer on port %d", port);
+      streamer = new WebcamStreamer(this.port, webcam, this.requestedFps, true);
     }
-    if (streamer == null) {
-      streamer = new WebcamStreamer(port, webcam, 0.5, true);
+
+    // webcam.addWebcamListener(this);
+
+    webcam.open();
+
+    worker = new Thread(new VideoProcessor(), String.format("%s-video-processor", getName()));
+    worker.start();
+  }
+
+  public void capture(String name, String type, Integer fps, Integer width, Integer height, Double quality) {
+    if (name == null) {
+      capture(0, type, fps, width, height, quality);
+    } else {
+      List<com.github.sarxos.webcam.Webcam> cams = com.github.sarxos.webcam.Webcam.getWebcams();
+      for (int i = 0; i < cams.size(); ++i) {

```

Comment:
Why are we capturing from all webcams on the system here?   We should specify which cameras should be captured from.. not all available cameras on the system..